scripts/resource/basic_discovery.rc

```diff
@@ -87,14 +87,15 @@ print_line("udp_sweep")
 run_single("use auxiliary/scanner/discovery/udp_sweep")
 run_single("run -j")
```

Comment:
Unfortunately this is still erroring out:

```
~/git/metasploit-framework │ bucket ?24  export METASPLOIT_RHOSTS='172.31.2.3/32,172.31.4.5/32'                                                                                 ✔ │ 4m 49s │ 3.0.5 Ruby │ 02:55:34 PM 
  
 ~/git/metasploit-framework │ bucket ?24  ./msfconsole --quiet --logger Stdout \                                                                                                          ✔ │ 3.0.5 Ruby │ 02:56:04 PM 
  --execute-command \
  "setg RHOSTS ${METASPLOIT_RHOSTS}; resource /home/gwillcox/git/metasploit-framework/scripts/resource/basic_discovery.rc; exit"
RHOSTS => 172.31.2.3/32,172.31.4.5/32
[*] Processing /home/gwillcox/git/metasploit-framework/scripts/resource/basic_discovery.rc for ERB directives.
[*] resource (/home/gwillcox/git/metasploit-framework/scripts/resource/basic_discovery.rc)> Ruby Code (20356 bytes)
THREADS => 15

============================================
starting discovery scanners ... stage 1
============================================


starting portscanners ...

udp_sweep
[-] Msf::OptionValidateError The following options failed to validate: RHOSTS
Module: db_nmap
Using Nmap with the following options: -n -PN -P0 -O -sSV 172.31.2.3/32 172.31.4.5/32
[*] Nmap: 'You requested a scan type which requires root privileges.'
[!] Running Nmap with sudo
[*] Nmap: Starting Nmap 7.93 ( https://nmap.org/ ) at 2023-02-15 14:56 CST
```

After speaking to colleagues about this I think we may want to just add in more `run_single('set RHOSTS XXXX')` calls here to fix this up.

Reply:
RHOSTS is set `setg RHOSTS ${METASPLOIT_RHOSTS};` here. So something like:

```
nmap_rhosts = datastore..gsub...
```

then set RHOSTS using `run_single("set RHOSTS #{nmap_hosts}")` explicitly?

Reply:
Did the following changes: (in Stage 1)
1. Created a temp variable for storing the space separated values.
2. Added run_single to set RHOSTS again.

Additional: (in Stage 2)
3. Changed auxiliary/admin/ftp/titanftp_xcrc_traversal to auxiliary/scanner/ftp/titanftp_xcrc_traversal as the earlier one does not exist anymore.